File: frontend/src/app/models/message.ts
export interface UserMin {
  id: number;
  name: string;
  role: string;
  profilePictureUrl?: string;
}

export interface MessageRead {
  id: number;
  senderId: number;
  receiverId: number;
  content: string;
  timestamp: string;
  listingId?: number;
  listingTitle?: string;
  isRead: boolean;
}

export interface ConversationRead {
  otherUser: UserMin;
  lastMessage: MessageRead;
  unreadCount: number;
}

File: backend-python/app/schemas/message.py
from datetime import datetime
from typing import Optional
from sqlmodel import SQLModel


class MessageSendPayload(SQLModel):
    receiverId: int
    content: str
    listingId: Optional[int] = None


class MessageReadDto(SQLModel):
    id: int
    senderId: int
    receiverId: int
    content: str
    timestamp: datetime
    listingId: Optional[int] = None
    listingTitle: Optional[str] = None
    isRead: bool


class UserMinDto(SQLModel):
    id: int
    name: str
    role: str
    profilePictureUrl: Optional[str] = None


class ConversationReadDto(SQLModel):
    otherUser: UserMinDto
    lastMessage: MessageReadDto
    unreadCount: int
